File: CustomDefaultEmailPlugin/src/com/ibm/ecm/extension/customDefaultEmail/CustomDefaultEmailPlugin.java
/*
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2020
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 */
package com.ibm.ecm.extension.customDefaultEmail;

import java.util.Locale;
import com.ibm.ecm.extension.*;

public class CustomDefaultEmailPlugin extends Plugin {

	public static String pluginId = "CustomDefaultEmail";
	private PluginResponseFilter[] responseFilters = null;

	@Override
	public String getId() {
		return "CustomDefaultEmailPlugin";
	}

	@Override
	public String getName(Locale locale) {
		return "Custom Default Email Plugin";
	}

	@Override
	public String getVersion() {
		return "3.0.9";
	}

	@Override
	public String getScript() {
		return "CustomDefaultEmailPlugin.js";
	}

	@Override
	public String getDojoModule() {
		return "customDefaultEmailPluginDojo";
	}

	@Override
	public String getConfigurationDijitClass() {
		return "customDefaultEmailPluginDojo.ConfigurationPane";
	}


	@Override
	public PluginResponseFilter[] getResponseFilters() {
		if (responseFilters == null) {
			responseFilters = new PluginResponseFilter[] { new CustomDefaultEmailPluginResponseFilter() };
		}
		return responseFilters;
	}

}

File: CustomDefaultEmailPlugin/src/com/ibm/ecm/extension/customDefaultEmail/WebContent/CustomDefaultEmailPlugin.js
/*
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2020
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 */
require(["dojo/aspect",
        "dojo/_base/declare",
        "dojo/_base/lang",
        "ecm/widget/dialog/EmailDialog",
        "ecm/model/Desktop"
], function(aspect, declare, lang, EmailDialog, Desktop) {

    //before _loadDefaults is called, add the subject and message values as parameters to loadDefaults method
    aspect.before(EmailDialog.prototype, "loadDefaults", lang.hitch(this, function(defaultValue, callback) {
        var subject = Desktop.pluginSettings && Desktop.pluginSettings.CustomDefaultEmail && Desktop.pluginSettings.CustomDefaultEmail.subject;
        var message = Desktop.pluginSettings && Desktop.pluginSettings.CustomDefaultEmail && Desktop.pluginSettings.CustomDefaultEmail.message;

        var value = {
            subject: subject,
            message: message
        };
        return [value, callback];

    }));
});
